sources/packages/web/src/components/aest/StudentApplications.vue

```diff
@@ -0,0 +1,290 @@
+     * @param pageCount page limit, if nothing passed then DEFAULT_PAGE_LIMIT
+     * @param sortField sort field, if nothing passed then StudentApplicationFields.status
+     * @param sortOrder sort oder, if nothing passed then DataTableSortOrder.ASC
+     */
+    const getStudentApplications = async (
+      page = DEFAULT_PAGE_NUMBER,
+      pageCount = DEFAULT_PAGE_LIMIT,
+      sortField = StudentApplicationFields.Status,
+      sortOrder = DataTableSortOrder.ASC,
+    ) => {
+      switch (props.clientType) {
+        case ClientIdType.Student:
+          applicationAndCount.value = await StudentService.shared.getAllStudentApplications(
+            page,
+            pageCount,
+            sortField,
+            sortOrder,
+          );
+          break;
+        case ClientIdType.AEST:
+          applicationAndCount.value = await ApplicationService.shared.getAllApplicationsForStudent(
+            props.studentId,
+            page,
+            pageCount,
+            sortField,
+            sortOrder,
+          );
+          break;
+      }
+    };
+
+    const reloadApplication = async () => {
+      await getStudentApplications();
+    };
+
+    onMounted(async () => {
+      if (props.clientType === ClientIdType.Student) {
+        const restrictions = await StudentService.shared.getStudentRestriction();
+        hasRestriction.value = restrictions.hasRestriction;
+      }
+      reloadApplication();
+    });
+
+    // pagination sort event callback
+    const paginationAndSortEvent = async (event: any) => {
+      loading.value = true;
+      currentPage.value = event?.page;
+      currentPageLimit.value = event?.rows;
+      await getStudentApplications(
+        event?.page,
+        event?.rows,
+        event?.sortField,
+        event?.sortOrder,
+      );
+      loading.value = false;
+    };
+
+    const openConfirmCancel = (id: number) => {
+      showModal.value = true;
+      selectedApplicationId.value = id;
+    };
+
+    const showHideCancelApplication = () => {
+      showModal.value = !showModal.value;
+    };
+
+    const goToApplication = (id: number) => {
+      return router.push({
+        name: StudentRoutesConst.STUDENT_APPLICATION_DETAILS,
+        params: {
+          id: id,
+        },
+      });
+    };
+
+    const getProgramYear = async (applicationId: number) => {
+      programYear.value = await ApplicationService.shared.getProgramYearOfApplication(
+        applicationId,
+      );
+    };
+
+    const editApplicaion = async (applicationId: number) => {
+      await getProgramYear(applicationId);
+      router.push({
+        name: StudentRoutesConst.DYNAMIC_FINANCIAL_APP_FORM,
+        params: {
+          selectedForm: programYear.value.formName,
+          programYearId: programYear.value.programYearId,
+          id: applicationId,
+        },
+      });
+    };
+    const confirmEditApplication = async (id: number) => {
+      if (await editApplicationModal.value.showModal()) {
```

Comment:
await editApplicationModal.value.showModal() can be assigned to a variable instead of directly using in if.

Reply:
why assign it as a variable?